azuredevops/resource_repository.go

```diff
@@ -81,33 +89,107 @@ func (r *repository) Trees() gitprovider.TreeClient {
 }
 
 func (r *repository) APIObject() interface{} {
-	//TODO implement me
-	panic("implement me")
+	return &r.r
 }
 
+// The internal API object will be overridden with the received server data.
 func (r *repository) Update(ctx context.Context) error {
-	//TODO implement me
-	panic("implement me")
-}
 
-func (r *repository) Reconcile(ctx context.Context) (actionTaken bool, err error) {
-	//TODO implement me
-	panic("implement me")
+	apiObj, err := r.g.UpdateRepository(ctx, git.UpdateRepositoryArgs{
+		Project:      r.r.Project.Name,
+		RepositoryId: r.r.Id,
+		NewRepositoryInfo: &git.GitRepository{
+			Name: r.r.Name,
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+	r.r = *apiObj
+	return nil
+}
+
+// Reconcile makes sure the desired state in this object (called "req" here) becomes
+// the actual state in the backing Git provider.
+//
+// If req doesn't exist under the hood, it is created (actionTaken == true).
+// If req doesn't equal the actual state, the resource will be updated (actionTaken == true).
+// If req is already the actual state, this is a no-op (actionTaken == false).
+//
+// The internal API object will be overridden with the received server data if actionTaken == true.
+
+func (r *repository) Reconcile(ctx context.Context) (bool, error) {
+
+	projectName := r.ref.GetIdentity()
+	repositoryID := r.ref.GetRepository()
+	apiObj, err := r.g.GetRepository(ctx, git.GetRepositoryArgs{
+		RepositoryId: &repositoryID,
+		Project:      &projectName,
+	})
+	if err != nil {
+		// Create if not found
+		if errors.Is(handleHTTPError(err), gitprovider.ErrNotFound) {
+			orgName := ""
```

Comment:
If you change:

`func newRepository(ctx *clientContext, apiObj git.GitRepository, ref gitprovider.RepositoryRef) *repository`

to 

`func newRepository(ctx *clientContext, apiObj git.GitRepository, ref gitprovider.OrgRepositoryRef) *repository`, 

that will make it cleaner and you could write `orgName := r.ref.Organization`